cmd/rpcdaemon/commands/zkevm_counters.go

```diff
@@ -36,53 +45,75 @@ type zkevmRPCTransaction struct {
 	To       *common.Address  `json:"to"`
 	From     *common.Address  `json:"from"`
 	Value    *hexutil.Big     `json:"value"`
-	V        *hexutil.Big     `json:"v"`
-	R        *hexutil.Big     `json:"r"`
-	S        *hexutil.Big     `json:"s"`
+	Data     hexutility.Bytes `json:"data"`
 }
 
 // Tx return types.Transaction from rpcTransaction
-func (tx *zkevmRPCTransaction) Tx() types.Transaction {
+func (tx *zkevmRPCTransaction) Tx(sr state.StateReader) (types.Transaction, error) {
 	if tx == nil {
-		return nil
+		return nil, nil
+	}
+
+	sender := common.HexToAddress(defaultSenderAddress)
+	if tx.From != nil {
+		sender = *tx.From
+	}
+	nonce := uint64(0)
+
+	ad, err := sr.ReadAccountData(sender)
+	if err != nil {
+		return nil, err
+	}
+	if ad != nil {
+		nonce = ad.Nonce
+	}
+
+	if tx.Value == nil {
+		// set this to something non nil
+		tx.Value = &hexutil.Big{}
 	}
 
 	gasPrice := uint256.NewInt(1)
 	if tx.GasPrice != nil {
 		gasPrice = uint256.MustFromBig(tx.GasPrice.ToInt())
 	}
+
+	var data []byte
+	if tx.Data != nil {
+		data = tx.Data
+	} else if tx.Input != nil {
+		data = tx.Input
+	} else if tx.To == nil {
+		return nil, fmt.Errorf("contract creation without data provided")
+	}
+
 	var legacy *types.LegacyTx
 	if tx.To == nil {
 		legacy = types.NewContractCreation(
-			uint64(tx.Nonce),
+			nonce,
 			uint256.MustFromBig(tx.Value.ToInt()),
 			uint64(tx.Gas),
 			gasPrice,
-			tx.Input,
+			data,
 		)
 	} else {
 		legacy = types.NewTransaction(
-			uint64(tx.Nonce),
+			nonce,
 			*tx.To,
 			uint256.MustFromBig(tx.Value.ToInt()),
 			uint64(tx.Gas),
 			gasPrice,
-			tx.Input,
+			data,
 		)
 	}
 
-	if tx.From != nil {
-		legacy.SetSender(*tx.From)
-	}
+	legacy.SetSender(sender)
 
-	if tx.V != nil && tx.R != nil && tx.S != nil {
-		// parse signature raw values V, R, S from local hex strings
-		legacy.V = *uint256.MustFromBig(tx.V.ToInt())
-		legacy.R = *uint256.MustFromBig(tx.R.ToInt())
-		legacy.S = *uint256.MustFromBig(tx.S.ToInt())
-	}
+	legacy.V = *uint256.MustFromHex(defaultV)
+	legacy.R = *uint256.MustFromHex(defaultR)
+	legacy.S = *uint256.MustFromHex(defaultS)
```

Comment:
do we not want to use the values passed in to the RPC if they are available and fall back to defaults if they're not there?